substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/code/SourceReferenceSupplier.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2021, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.code;
+
+import com.oracle.svm.core.hub.DynamicHub;
+
+final class SourceReferenceSupplier {
```

Comment:
As already discussed with @ivan-ristovic, this uses the `overlayTarget` mechanism that we want to get rid of.

Let's wait with merging this until we have Java 8 removed from substratevm. Then is fallback class will become obsolete.

Reply:
Alternatively you can use `com.oracle.svm.util.ModuleSupport#getModuleName` to avoid introducing two variants of `com.oracle.svm.core.code.SourceReferenceSupplier`.

Reply:
That is true, however I could not remove the need for two separate constructors for the `StackTraceElement`, one of which is only present in JDK9+ (with module-related arguments).